freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UsersyncPoller.java

```diff
@@ -0,0 +1,110 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.security.InternalCrnBuilder;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.freeipa.api.v1.freeipa.user.model.SyncOperationStatus;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.entity.UserSyncStatus;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UmsEventGenerationIds;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@Service
+public class UsersyncPoller {
+    @VisibleForTesting
+    static final String INTERNAL_ACTOR_CRN = new InternalCrnBuilder(Crn.Service.IAM).getInternalCrnForServiceAsString();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UsersyncPoller.class);
+
+    @Inject
+    private ThreadBasedUserCrnProvider threadBasedUserCrnProvider;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private UserSyncStatusService userSyncStatusService;
+
+    @Inject
+    private UserService userService;
+
+    @Inject
+    private UmsEventGenerationIdsUtil umsEventGenerationIdsUtil;
+
+    @Value("${freeipa.syncoperation.poller.enabled:true}")
```

Comment:
I am not sure how easy it is to do, but instead of having the notion of enabled and disabled in the poller you can instead just instantiate the poller if the sync poller is enabled.

Reply:
I can investigate that for the future. It should be doable in one of the Configuration classes to either explicitly create/not create the bean